src/renderer/components/+preferences/preferences-navigation/preferences-navigation.tsx

```diff
@@ -13,30 +13,58 @@ import type {
 import preferenceNavigationItemsInjectable from "./preference-navigation-items.injectable";
 
 import { observer } from "mobx-react";
+import { Icon } from "../../icon";
 
 interface Dependencies {
   navigationItems: IComputedValue<PreferenceNavigationItem[]>;
 }
 
 const NonInjectedPreferencesNavigation = ({
   navigationItems,
-}: Dependencies) => (
-  <Tabs
-    className="flex column"
-    scrollable={false}
-    onChange={(item: PreferenceNavigationItem) => item.navigate()}
-  >
-    <div className="header">Preferences</div>
-
-    {navigationItems.get().map((item) => (
+}: Dependencies) => {
+  const generalNavItems = navigationItems.get().filter(item => !item.fromExtension);
+  const extensionNavItems = navigationItems.get().filter(item => item.fromExtension);
+
+  function renderTab(item: PreferenceNavigationItem) {
+    return (
       <PreferencesNavigationTab
         key={item.id}
         item={item}
         data-testid={`tab-link-for-${item.id}`}
       />
-    ))}
-  </Tabs>
-);
+    );
+  }
+
+  return (
+    <Tabs
+      className="flex column"
+      scrollable={false}
+      onChange={(item: PreferenceNavigationItem) => item.navigate()}
+    >
+      <div className="header">Preferences</div>
+
+      {generalNavItems.map(renderTab)}
+
+      {extensionNavItems.length > 0 && (
+        <div data-testid="extension-settings">
+          <hr/>
+          <div className="header flex items-center">
+            <Icon
+              material="extension"
+              smallest
+              className="mr-3"
+            />
+            {" "}
+            Extensions
+          </div>
+          <div>
+            {extensionNavItems.map(renderTab)}
+          </div>
+        </div>
+      )}
+    </Tabs>
+  );
+};
```

Comment:
This is code that should be refactored in the future to way that we just loop items through. This is because there is no need to hard-code stuff like "Preferences", "Extensions" and order of them. If I want to add third one, I shouldn't need to change this file at all.

Reply:
Refactored to much simpler version of it. Created one more level of abstraction by introducing `navigation groups` which contain related preference links.